### ISSUE TITLE
Lost pipeline output data with Azure blob storage (returns 404)

### ISSUE THREAD

**Issue by itssimon** [pachyderm/pachyderm#3722]
**What happened?**

My pipeline has produced an output file which is not accessible subsequently. This happened a few times sporadically, but unfortunately I can't reliable reproduce it (the pipeline runs for 6 hours too).

So the file is listed as part of the output commit of my pipeline:

```
$ pachctl list-file my_pipeline af60788536d5420aa0a4ee7f01f8f4d0 /tsv/ROA_759986.tsv
COMMIT                           NAME                TYPE COMMITTED    SIZE
af60788536d5420aa0a4ee7f01f8f4d0 /tsv/ROA_759986.tsv file 37 hours ago 11.63KiB
```

But getting the file fails with BlobNotFound (404):

```
$ pachctl get-file my_pipeline af60788536d5420aa0a4ee7f01f8f4d0 /tsv/ROA_759986.tsv
storage: service returned error: StatusCode=404, ErrorCode=BlobNotFound, ErrorMessage=The specified blob does not exist.
RequestId:371e303f-c01e-0000-4e2a-09ee57000000
Time:2019-05-13T01:22:03.3705893Z, RequestId=371e303f-c01e-0000-4e2a-09ee57000000, QueryParameterName=, QueryParameterValue=
```

This also means that any pipeline relying on this fails.

I have checked the logs of the pipeline creating this file and it logs `finished uploading output after 211.136662ms` for this particular file/datum. No errors there.

**What you expected to happen?**

The output file can be retrieved.

**How to reproduce it**

It happens sporadically and is hard to reproduce in a minimal example.

**Environment**
- Kubernetes version: 1.13.5
- Pachyderm CLI version: 1.8.6
- Cloud provider: Azure AKS

**Comment by JoeyZwicker** [comment]
HI @itssimon We think we might have a bead on what could cause issues like this. We're still trying to get a clear resolution path and will update you when we have one in place.

**Comment by JoeyZwicker** [comment]
I believe the situation that could lead to this issue has been resolved 1.9.0-rc3. Unfortunately, it means that there's some corrupted data in your blob store currently that's not really fixable. @brycemcanally Is this correct or is there actually a way to fix the the bucket?

@itssimon If you hit this issue again on a fresh deploy of 1.9 or later, please let us know. Sorry you hit this issue!
